### PR TITLE
docs: add a note about compilation options

### LINKED ISSUE #6371: meson: should default to LTO again
### Submission type

 - Bug report

### systemd version the issue has been seen with

> v234

### Used distribution

> Arch Linux

### In case of bug report: Expected behaviour you didn't see

> Similar binary sizes compared to auto-tools build.

### In case of bug report: Unexpected behaviour you saw

> Building with meson increases installed package size by about 10MB (+50%). This hits binary sizes for executables and libraries.

### In case of bug report: Steps to reproduce the problem

> Build with meson and auto-tools and compare sizes.

### PR DESCRIPTION
**Pull request by keszybz**
Closes #6371.

### CODE REVIEW

File: docs/DISTRO_PORTING.md [hunk 1]
```diff
@@ -31,6 +31,20 @@ distribution:
...
+
+The default configuration does not enable any optimization or hardening
+options. This is suitable for development and testing, but not for end-user
+installations.
+
+For deployment, optimization (`-O2` or `-O3` compiler options), link time
+optimization (`-Db_lto=true` meson option), hardening (e.g.
+`-D_FORTIFY_SOURCE=2`, `-fstack-protector-strong`, `-fstack-clash-protection`,
+`-fcf-protection`, `-pie` compiler options, and `-z relro`, `-z now`,
+`--as-needed` linker options), are recommended. The most appropriate set of
```
Comment (poettering) [inline docs/DISTRO_PORTING.md]: hmm, this comma is too much, no?

**Comment by yuwata** [comment]
BTW, is it necessary to wait CIs?

**Comment by poettering** [comment]
nope, not really, but I wanted to give @keszybz time to comment on my comma comment ;-)

File: docs/DISTRO_PORTING.md (hunk 1 above)
Comment (keszybz) [reply docs/DISTRO_PORTING.md]: Updated.

**Comment by keszybz** [comment]
I'll merge since this is just docs and the review was (mostly) positive.

**Pull request merged**